hercules-ci-agent/hercules-ci-agent/Hercules/Agent/Config.hs

```diff
@@ -59,6 +77,10 @@ determineDefaultApiBaseUrl = do
 defaultApiBaseUrl :: Text
 defaultApiBaseUrl = "https://hercules-ci.com"
 
+defaultBaseDirectory :: IO FilePath
+defaultBaseDirectory =
+  System.Directory.getAppUserDataDirectory "hercules-ci-agent"
```

Comment:
Why not default to `/var/lib/hercules-ci-agent`? It would make documentation easier, it's always there unless you explicitly want something else.

Reply:
By defaulting to `~/.hercules-ci-agent`, we simplify the setup instructions for someone who wants to assess the product using their local machine. It's just one little thing that may help a (potential) customer achieve a working setup quickly.

Doc suggestion:
> The path below which all configuration and state are stored. The default values for a number of other locations derive from this option.
> Default value: `~/.hercules-ci-agent`. Deployments via NixOS, NixOps and nix-darwin override the default to `/var/lib/hercules-ci-agent`.

Reply:
At the moment deploying without Nix is still painful (and it doesn't seem we can really ever make it simpler, as it's getting more complex), so is that really a requirement we should have?

It seems to be it would be easier to come up with arion setup for assessing and keeping the complexity of directory defaults low.